volume/btrfs/btrfs.go

```diff
@@ -317,8 +318,41 @@ func (v *BtrfsVolume) isSnapshot(rawLabel string) bool {
 	return strings.HasPrefix(rawLabel, v.getSnapshotPrefix())
 }
 
+// writeSnapshotInfo writes metadata about a snapshot
+func (v *BtrfsVolume) writeSnapshotInfo(label string, info *volume.SnapshotInfo) error {
```

Comment:
There's a bit of copy/paste code with the same impls of writeSnapshotInfo() and SnapshotInfo() across the different volumes.  I see that there is a lot of copy/paste code of generic methods like rawSnapshotLabel(), getSnapshotPrefix(), etc.

Seems like there should be some way to consolidate more of that code so it's not duplicated in so much.
In Java or C++, something like an AbstractBaseClass would work, but for GO you would have to make more of the methods public. Another alternative, is to consolidate all of the common code into something like a 'volume helper' class which is injected into each of Volume implementations.